tests/manage/test_pvc_disruptive.py

```diff
@@ -21,43 +61,171 @@ class BaseDisruption(ManageTest):
     pvc_obj = None
     storage_type = None
 
+    def create_pvcs(self):
+        """
+        Create PVCs
+
+        Returns:
+            list: List of PVC objects
+        """
+        # Parameters for PVC yaml as dict
+        pvc_data = load_yaml_to_dict(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = self.sc_obj.name
+        pvc_data['metadata']['name'] = helpers.create_unique_resource_name('test', 'pvc')
+
+        # Create 5 PVCs
+        pvc_objs = create_multiple_pvc(5, pvc_data)
+        return pvc_objs
+
+    def create_pods(self, pvc_objs_list):
+        """
+        Create Pods.
+        A pod will be created for each PVC in 'pvc_objs_list'.
+
+        Args:
+            pvc_objs_list (list): List of PVC objects
+
+        Returns:
+            list: List of Pod objects
+        """
+        pod_objs = []
+        for pvc_obj in pvc_objs_list:
+            pod_obj = helpers.create_pod(
+                interface_type=constants.CEPHBLOCKPOOL, pvc_name=pvc_obj.name, wait=False, namespace=self.namespace
+            )
+            pod_objs.append(pod_obj)
+        return pod_objs
+
     def disruptive_base(self, operation_to_disrupt, resource_to_delete):
         """
         Base function for PVC disruptive tests
+        Deletion of 'resource_to_delete' will be introduced while 'operation_to_disrupt' is progressing
         """
+        # Fetch the number of Pods and PVCs
+        initial_num_of_pods = len(pod.get_all_pods(namespace=self.namespace))
+        initial_num_of_pvc = len(get_all_pvcs(namespace=self.namespace)['items'])
+
+        executor = ThreadPoolExecutor(max_workers=1)
+
         DISRUPTION_OPS.set_resource(resource=resource_to_delete)
-        self.pvc_obj = helpers.create_pvc(
-            sc_name=self.sc_obj.name, wait=False
-        )
+
+        # Start creation of multiple PVCs
+        bulk_pvc_create = executor.submit(self.create_pvcs)
+
         if operation_to_disrupt == 'create_pvc':
-            DISRUPTION_OPS.delete_resource()
-        self.pvc_obj.reload()
-        assert self.pvc_obj.ocp.wait_for_resource(
-            condition=constants.STATUS_BOUND, resource_name=self.pvc_obj.name, timeout=120
-        )
-
-        self.pod_obj = helpers.create_pod(
-            interface_type=constants.CEPHBLOCKPOOL, pvc_name=self.pvc_obj.name, wait=False
-        )
+            # Ensure PVCs are being created before deleting the resource
+            for count in range(10):
```

Comment:
I think it would be better to avoid use TimeoutSampler. This is from my head and didn't actually test it, so could be there are some fine tunings that need to be done for it to work:

```python
for sample in TimeoutSampler(
    30, 1, get_all_pvcs, self.namespace
):
    if len(sample['items']) > initial_num_of_pvc:
        DISRUPTION_OPS.delete_resource()
```

Reply:
Done